mappings/net/minecraft/util/registry/RegistryKey.mapping

```diff
@@ -55,3 +57,5 @@ CLASS net/minecraft/class_5321 net/minecraft/util/registry/RegistryKey
 		ARG 0 registry
 	METHOD method_39155 (Lnet/minecraft/class_5321;Lnet/minecraft/class_2960;)Lnet/minecraft/class_5321;
 		ARG 1 id
+	METHOD method_39752 matchingRegistry (Lnet/minecraft/class_5321;)Ljava/util/Optional;
```

Comment:
This feels strange, but it should be fine. I would be inclined to name it `ifMatchingRegistry` if that did not imply too heavily that it would act like `Optional.ifPresent`.

Reply:
Notice `ifPresent` is a functional method that only executes the given consumer conditionally; this one is more like `filter` `map` of `Optional`, and IMO the current name is better.

Reply:
To me, it looks like a safe cast (I got an idea from the `Prism` optic, but `Optional` instead of `Either`). In its only one usage, `key.method_39752(ref)` casts from `RegistryKey<?>` to `RegistryKey<E>` if the `key` belongs to the `ref`, and yields `Optional.empty()` otherwise. Is `tryCast` a good name?

Reply:
Ahhh! That makes 100% sense.